lib/ftp.c

```diff
@@ -3500,7 +3501,11 @@ static CURLcode ftp_range(struct connectdata *conn)
     }
     else {
       /* X-Y */
-      data->req.maxdownload = (to - from) + 1; /* include last byte */
+      totalsize = to-from;
+      if(totalsize == CURL_OFF_T_MAX)
+        /* this is too big to increase, so bail out */
```

Comment:
I would get rid of this comment it's implied. also you can declare totalsize in this block. also since we're here this could use a check to make sure from <= to

Reply:
This code is basically a copy of the code from https://github.com/curl/curl/blob/master/lib/file.c#L166 that @bagder wrote; should we be updating both places?

Reply:
We could, but I'd rather not lump a change of file.c into this fix as it isn't really related to this FTP range integer overflow...

Reply:
The reason I ask is that you made a change to file_range to fix a bug, file_range has the comment
```
 /*
  Check if this is a range download, and if so, set the internal variables
  properly. This code is copied from the FTP implementation and might as
  well be factored out.
 */
```
and the ftp_range function still has the issue. So my assumption is that any fixes (or markups) that need to be made in one place should be made in the other as well.

Reply:
Let's turn `file_range()` into `Curl_range()` and make both the FILE and FTP code use it.

The only difference I can spot is that `ftp_range()` sets `ftpc->dont_check = TRUE` when a range has been set, so we should make sure to set that separately.

What do you think?

Reply:
Happy with that. Do you want to run with that or shall I?

Reply:
Please go ahead, I have a few other tasks to work on anyway! =)